### ISSUE TITLE
WebDAV sync doesn't work if there's a space in the folder name.

### ISSUE THREAD

**Issue by r4dian** [orgzly-revived/orgzly-android-revived#380]
When you have a folder named `org files`:

1. If you set the URL in Orgzly as `https://███████.cloud/remote.php/dav/files/████%40████████.co.uk/org files` (with the space) then "Test Connection" fails
2. If you set the URL in Orgzly as `https://███████.cloud/remote.php/dav/files/████%40████████.co.uk/org%20files` (with %20 where the space would go) then "Test Connection" succeeds & finds the correct amount of note books, then when you try to sync each one appears, but gets error 404 & can't sync.
3. If you leave the folder off the URL and tick the use subfolders option, the result is the same as 2, and you can see the URL for each file is using `org%20files/` as the folder name.

- [X] I have searched for existing issues that may be the same as or related to mine.

Please complete the following information:

    Device: Sony Xperia 10 III (aka XQ-BT52)
    Android Version: 13
    Orgzly Revived Version: 1.8.28 (via f-droid )
    Sync repository type (directory/Dropbox/WebDAV/Git): WebDAV

I missed several updates cos I didn't know f-droid had newer versions than one I was on, so unfortunately don't know in which version this broke.

![image](https://github.com/user-attachments/assets/59405954-c2e8-481c-92c3-345a101e6a80)

**Comment by amberin** [comment]
Thanks for reporting, if this used to work then it's pretty serious. I'll look into it ASAP.

**Comment by r4dian** [comment]
I last used it this way in Orgzly (RIP) rather than Orgzly Revived & have been using it thorough the Nextcloud app since.

**Comment by Forge36** [comment]
There's a mention of fixing a bug with spaces in URLs with a dependency:

https://github.com/thegrizzlylabs/sardine-android/commit/5d7883cef19b2753c6e508a2febb44084a7ea8c2

**Comment by amberin** [comment]
I am able to reproduce this without using folders, just using a space in a file name triggers this issue for me. The file cannot be loaded from the repository, probably due to mishandling of the URI somewhere. I'll look into it.

**Comment by amberin** [comment]
@r4dian So the server is backed by Nextcloud? I'm seeing the same issue with a Nextcloud instance, and I believe the problem is not Orgzly, but rather that Nextcloud hands out repository URLs where the @ is correctly encoded (into %40), but then it advertises the DAV resources (files) with an unencoded @ in the URLs. In other words, Nextcloud is being inconsistent, and there is no sane way for Orgzly to work around that.

I got rid of the problem by using @ instead of %40 in the Orgzly WebDAV repository URL.

If we are having the same issue, then the spaces are not the problem -- they should always be encoded in a URL, so your scenario no. 1 is just wrong.

**Comment by lckarssen** [comment]
I've got a (Nextcloud 32) webDAV path that contains brackets, i.e. `(` and `)`, and a comma, as well as spaces. I have replaced all, except the spaces, with their non-encoded counterparts but the sync still fails. The Orgzly connection test correctly sees there is a single Org file in that directory, but syncing the file fails, with the full URL repeated where the file name should have been.

**Comment by amberin** [comment]
Yeah, the repeat of the full URL is due to failed string substitution. Most problems we've seen like this could be worked around by encoding all such characters when setting up the repo. If that does NOT solve your issue, then that is very interesting, because we've said that we should encode the user-provided URL before storing it. So, please try entering a fully-encoded URL when setting up the repo, and report whether that works. 🙏

**Comment by amberin** [comment]
Sorry, my previous comment was based on misremembering. Looking at the conversation here and in #420, I now remember that encoding the characters in the stored repo URL is not a solution. So I'm still not sure how to handle this. Except some hacky heuristics which tries different encodings if the file name ends up looking "wrong".

**Comment by lckarssen** [comment]
I had a quick look at this again and in my case, removing the `,` from the Nextcloud directory name solved the issue. So it seems that some characters are more problematic than others:
- spaces are fine if encoded as `%20` in the sync URL
- brackets/braces `(` and `)` are fine if *not* encoded
- `@` is fine if *not* encoded (from https://github.com/orgzly-revived/orgzly-android-revived/issues/420#issuecomment-2523974976)
- commas `,` are not fine at all, neither unencoded, nor encoded (as `%2C`)

😕

**Comment by lckarssen** [comment]
Another quick test: when browsing `davs://MY_HOSTNAME/remote.php/dav/files/MY_USERNAME/`  from the Nautilus file manager, the directory (with the `,` in its name) and the files below it are presented as expected. This gives me the impression that the problem is not in the Nextcloud webDAV implementation, but rather something on the client side. 

Having said that, I have no clue where to look now. I have no knowledge of Android development, nor any experience developing WebDAV stuff. 

Also, I'm still mystified by the fact that if the URL contains a "forbidden" character, Orgzly still finds the correct number of Org files when testing the connection. As if listing the files works fine, but actually opening them fails (probably because of it taking a different code path)... 🤔

**Comment by amberin** [comment]
It's most likely a limitation in the WedDAV library that we use. But needs to be examined.
